### PR TITLE
chore: remove rewards app, update permit types

### PR DESCRIPTION
**Pull request by Keyrxng**
Related to https://github.com/ubiquity/pay.ubq.fi/pull/185

- removed unused rewards app js kept the css
- moved over the shared abis folder
- copy/pasted the updated permit types and replaced ``TxData`` 
- the type ``Permit`` that should be generated from the DB schema still exists in ``audit.ts``

### CODE REVIEW

**Comment by molecula451** [comment]
looking good, would help some testing at https://f9286cfc.audit-ubq-fi.pages.dev/

**Review by rndquu — APPROVED** [review]

**Review by molecula451 — APPROVED** [review]
gitcoindev is fixing knip, tho

File: .cspell.json [hunk 1]
```diff
@@ -6,11 +6,13 @@
   "language": "en",
   "words": [
     "binkey",
+    "Knip",
     "binsec",
     "chainlist",
     "cirip",
     "dataurl",
     "devpool",
+    "xrandomwallet",
```
Comment (gitcoindev) [inline .cspell.json:15]: I was curious where it came from, 0xrandomwallet!

**Review by gitcoindev — APPROVED** [review]
Approved, let's skip Knip CI failures till fixed this week.

File: .cspell.json (hunk 1 above)
Comment (Keyrxng) [reply .cspell.json:15]: In the readme, it had me in the first half ngl

```
{
  "WALLET": "0xrandomwallet",
  "REPO": "https://github.com/repo/repo",
  "PAT": "YOUR_PAT"
}
```

**Pull request merged**